### ISSUE TITLE
TypeInitializationEx: System.DllNotFoundException: Unable to load shared library '/home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/lib/linux/x86_64/libgit2-6311e88.so

### ISSUE THREAD

**Issue by VidyaKukke** [dotnet/Nerdbank.GitVersioning#676]
I seem to be getting an error with nerdbank versionioning (2.3.11-beta) when running on:
ImageOS=ubuntu20
ImageVersion=20211017.0

Should I consider moving to a newer version to fix or is there something more fundamentally wrong?

Thanks,
Vidya

2021-10-20T00:51:17.7727087Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: The "Nerdbank.GitVersioning.Tasks.GetBuildVersion" task failed unexpectedly. [/<somepath>/someproj.csproj.csproj]
2021-10-20T00:51:17.7729863Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7732741Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:  ---> System.DllNotFoundException: Unable to load shared library '/home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/lib/linux/x86_64/libgit2-6311e88.so' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libssl.so.1.0.0: cannot open shared object file: No such file or directory [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7735338Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.InternalLoadUnmanagedDllFromPath(String unmanagedDllPath) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7739290Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.LoadUnmanagedDllFromPath(String unmanagedDllPath) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7741409Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at MSBuildExtensionTask.ContextAwareTask.CustomAssemblyLoader.LoadUnmanagedDll(String unmanagedDllName) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7744064Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7746103Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init() [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7748682Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.LoadNativeLibrary() [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7750580Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor() [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7752246Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    --- End of inner exception stack trace --- [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7754469Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_opts(Int32 option, UInt32 level, String path) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7756303Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.Core.Proxy.git_libgit2_opts_set_search_path(ConfigurationLevel level, String path) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7758470Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at LibGit2Sharp.GlobalSettings.SetConfigSearchPaths(ConfigurationLevel level, String[] paths) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7760455Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.GitExtensions.OpenGitRepo(String pathUnderGitRepo) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7762490Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.VersionOracle.Create(String projectDirectory, String gitRepoDirectory, ICloudBuild cloudBuild, Nullable`1 overrideBuildNumberOffset, String projectPathRelativeToGitRepoRoot) [<somepath>/someproj.csprojCommon.csproj]
2021-10-20T00:51:17.7764400Z /home/vsts/.nuget/packages/nerdbank.gitversioning/2.3.11-beta/build/Nerdbank.GitVersioning.targets(63,5): error MSB4018:    at Nerdbank.GitVersioning.Tasks.GetBuildVersion.ExecuteInner() [<somepath>/someproj.csprojCommon.csproj]

**Comment by AArnott** [comment]
Yes, please try upgrading to v2.4 as it replaces use of libgit2sharp (with its SSL native dependencies that are so hard to satisfy across linux distros) with a managed git implementation.
